flixel/addons/weapon/FlxWeapon.hx

```diff
@@ -55,72 +56,58 @@ class FlxTypedWeapon<TBullet:FlxBullet>
 	public static inline var BULLET_SOUTH_WEST:Int = 135;
 
 	/**
-	 * Internal name for this weapon (i.e. "pulse rifle")
+	 * Internal name for this weapon (e.g., `"pulse rifle"`).
 	 */
 	public var name:String;
 
 	/**
-	 * The FlxGroup into which all the bullets for this weapon are drawn. This should be added to your display and collision checked against it.
+	 * The `FlxGroup` from which all the bullets for this weapon are drawn. This should be added to your display and collision-checked against it.
 	 */
 	public var group(default, null):FlxTypedGroup<TBullet>;
 
-	// Internal variables, use with caution
-	public var nextFire:Int = 0;
-
 	/**
-	 * The delay in milliseconds (ms) between which each bullet is fired, set to zero to clear. By default there is no rate, as it can be controlled by FlxControl.setFireButton.
-	 * However if you are firing using the mouse you may wish to set a firing rate.
+	 * The delay in milliseconds (ms) between which each bullet is fired. Default is 0, which means there is no delay.
 	 */
 	public var fireRate:Int = 0;
 
 	/**
-	 * When a bullet goes outside of this bounds it will be automatically killed, freeing it up for firing again.
-	 * TODO - Needs testing with a scrolling map (when not using single screen display)
-	 */
-	public var bounds:FlxRect;
-
-	/**
-	 * Only accessible when fireFrom is "PARENT"
+	 * A timer used whenever the weapon is fired, with its duration set to `fireRate`.
 	 */
-	public var parent(default, null):FlxSprite;
+	public var fireTimer:FlxTimer;
 
 	/**
-	 * If true, when fired the bullet direction is based on parent sprites facing value (up/down/left/right)
-	 */
-	public var useParentDirection:Bool;
```

Comment:
We can't just remove vars, we deprecate them first but leave them functioning until the next major version